runners/flink/src/test/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslationsTest.java

```diff
@@ -122,4 +156,206 @@ public void testBeamKinesisSchemaFutureOccurredAtTimestamp() throws Exception {
+    byte[] payloadBytes = new ObjectMapper().writeValueAsBytes(payload);
+    RunnerApi.Pipeline pipeline = createPipeline(id, payloadBytes);
+
+    // test
+    portableTranslations.translateKafkaInput(id, pipeline, streamingContext);
+
+    // verify
+    ArgumentCaptor<FlinkKafkaConsumer011> kafkaSourceCaptor =
+        ArgumentCaptor.forClass(FlinkKafkaConsumer011.class);
+    ArgumentCaptor<String> kafkaSourceNameCaptor = ArgumentCaptor.forClass(String.class);
+    verify(streamingEnvironment)
+        .addSource(kafkaSourceCaptor.capture(), kafkaSourceNameCaptor.capture());
+    Assert.assertEquals(
+        WindowedValue.class, kafkaSourceCaptor.getValue().getProducedType().getTypeClass());
+    Assert.assertTrue(kafkaSourceNameCaptor.getValue().contains(topicName));
+  }
+
+  @Test
+  public void testRequiredPropertiesForKafkaSink() throws JsonProcessingException {
+
+    LyftFlinkStreamingPortableTranslations portableTranslations =
+        new LyftFlinkStreamingPortableTranslations();
+    String id = "1";
+    String topicName = "kinesis_to_kafka";
+
+    Properties kakfaProducerProps = new Properties();
+    kakfaProducerProps.put("bootstrap.servers", "test-hdd.lyft.com");
+
+    ImmutableMap<String, Object> payload =
+        ImmutableMap.<String, Object>builder()
+            .put("topic", topicName)
+            .put("properties", kakfaProducerProps)
+            .put("username", "read_kinesis_to_kafka")
+            .put("password", "abcde1234")
+            .build();
+
+    byte[] payloadBytes = new ObjectMapper().writeValueAsBytes(payload);
+    RunnerApi.Pipeline pipeline = createPipeline(id, payloadBytes);
+
+    DataStream mockDataStream = mock(DataStream.class);
+    SingleOutputStreamOperator mockStreamOperator = mock(SingleOutputStreamOperator.class);
+    DataStreamSink mockStreamSink = mock(DataStreamSink.class);
+
+    when(mockStreamOperator.addSink(any())).thenReturn(mockStreamSink);
+    when(mockDataStream.transform(anyString(), any(), any())).thenReturn(mockStreamOperator);
+    when(streamingContext.getDataStreamOrThrow("fake_pcollection_id")).thenReturn(mockDataStream);
+
+    portableTranslations.translateKafkaSink(id, pipeline, streamingContext);
+
+    verify(streamingContext).getDataStreamOrThrow("fake_pcollection_id");
+    verify(mockDataStream).transform(anyString(), any(), any());
+    ArgumentCaptor<FlinkKafkaProducer011> kafkaSinkCaptor =
+        ArgumentCaptor.forClass(FlinkKafkaProducer011.class);
+    ArgumentCaptor<String> kafkaSinkNameCaptor = ArgumentCaptor.forClass(String.class);
+    verify(mockStreamSink).name(kafkaSinkNameCaptor.capture());
+    verify(mockStreamOperator).addSink(kafkaSinkCaptor.capture());
+
+    Assert.assertTrue(kafkaSinkNameCaptor.getValue().contains(topicName));
+    Assert.assertEquals(FlinkKafkaProducer011.class, kafkaSinkCaptor.getValue().getClass());
+  }
+
+  @Test
+  public void testKafkaSinkForLocalEnvironment() throws JsonProcessingException {
```

Comment:
Why is this test necessary in addition to the above test? It looks like this largely tests the same, other than using a different bootstrap server. If we want to keep this, maybe remove the duplicate test code.

Reply:
Good point! Will refactor the code to avoid bloating the tests.

Reply:
Refactored the code.